cmd/snap-update-ns/update.go

```diff
@@ -71,7 +76,7 @@ func executeMountProfileUpdate(upCtx MountProfileUpdateContext) error {
 
 	var changesMade []*Change
 	for _, change := range changesNeeded {
-		synthesised, err := change.Perform(as)
+		synthesised, err := change.Perform(upCtx)
```

Comment:
Consider adding change to filter out any unknown kinds of desired changes before earlier in executeMountProfileUpdate to make adding new kinds of changes simpler/safer in terms of compatibility if ever needed again